### PR TITLE
Add tabs_overlay

### PR DESCRIPTION
**Pull request by nt4f04uNd**
Moves https://github.com/nt4f04uNd/tabs_overlay to this repository.

<details>
<summary>Android video</summary>

https://user-images.githubusercontent.com/39104740/146440570-ba140add-0411-454a-9456-34d6d62e9fc8.mp4

</details>

<details>
<summary>iOS video</summary>

https://user-images.githubusercontent.com/39104740/149370283-aa5ecd0a-1f51-44fe-8e09-430e0c38051e.mp4

</details>

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: tabs_overlay/android/app/src/main/kotlin/com/example/tabs_overlay/MainActivity.kt [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    fun bindFlutterView(view: FlutterView) {
+        flutterView = view
+    }
+
+    override fun dispatchTouchEvent(ev: MotionEvent?): Boolean {
+        flutterView.dispatchTouchEvent(ev)
+        return super.dispatchTouchEvent(ev)
+    }
+
+}
```
Comment (Piinks) [inline tabs_overlay/android/app/src/main/kotlin/com/example/tabs_overlay/MainActivity.kt:94]: Nit: Add a newline here

File: tabs_overlay/android/app/src/main/kotlin/com/example/tabs_overlay/PageViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+class PageViewModel : ViewModel() {
+    private val _index = MutableLiveData<Int>()
+    val text: LiveData<String> = Transformations.map(_index) {
+        "Android tab $it"
+    }
+
+    fun setIndex(index: Int) {
+        _index.value = index
+    }
+}
```
Comment (Piinks) [inline tabs_overlay/android/app/src/main/kotlin/com/example/tabs_overlay/PageViewModel.kt:17]: Same here

File: tabs_overlay/android/app/src/main/kotlin/com/example/tabs_overlay/PlaceholderFragment.kt [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        @JvmStatic
+        fun newInstance(sectionNumber: Int): PlaceholderFragment {
+            return PlaceholderFragment().apply {
+                arguments = Bundle().apply {
+                    putInt(ARG_SECTION_NUMBER, sectionNumber)
+                }
+            }
+        }
+    }
+}
```
Comment (Piinks) [inline tabs_overlay/android/app/src/main/kotlin/com/example/tabs_overlay/PlaceholderFragment.kt:59]: Same here

File: tabs_overlay/android/app/src/main/kotlin/com/example/tabs_overlay/SectionsPagerAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+    override fun getPageTitle(position: Int): CharSequence? {
+        return TAB_TITLES[position]
+    }
+
+    override fun getCount(): Int {
+        // Show 2 total pages.
+        return 2
+    }
+}
```
Comment (Piinks) [inline tabs_overlay/android/app/src/main/kotlin/com/example/tabs_overlay/SectionsPagerAdapter.kt:34]: Same here

File: tabs_overlay/android/app/src/main/res/layout/activity_main.xml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        </com.google.android.material.appbar.AppBarLayout>
+
+        <androidx.viewpager.widget.ViewPager
+            android:id="@+id/view_pager"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            app:layout_behavior="@string/appbar_scrolling_view_behavior" />
+
+    </androidx.coordinatorlayout.widget.CoordinatorLayout>
+</com.example.tabs_overlay.CustomFrameLayout>
```
Comment (Piinks) [inline tabs_overlay/android/app/src/main/res/layout/activity_main.xml:50]: Here too :)

File: tabs_overlay/android/app/src/main/res/layout/fragment_main.xml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="@+id/view"
+        android:layout_marginBottom="40dp"
+        app:layout_constraintStart_toStartOf="@+id/view"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:text="John Doe" />
+
+</androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (Piinks) [inline tabs_overlay/android/app/src/main/res/layout/fragment_main.xml:31]: Same same

File: tabs_overlay/android/app/src/main/res/values/dimens.xml [hunk 7]
```diff
@@ -0,0 +1,5 @@
+<resources>
+    <!-- Default screen margins, per the Android Design guidelines. -->
+    <dimen name="appbar_padding">16dp</dimen>
+    <dimen name="appbar_padding_top">8dp</dimen>
+</resources>
```
Comment (Piinks) [inline tabs_overlay/android/app/src/main/res/values/dimens.xml:5]: And here

**Review by Piinks — COMMENTED** [review]
This is really nice, thank you for adding it!

Can you add a smoke test of some kind that just validates this runs? I'd like to see if we can add it to flutter/tests so we do not break this.

Often, when we create a tool like this, when we need it and call it up, it's broken and out of date. If we can add a simple smoke test and connect it to flutter/tests, that will ensure it never happens and it receives regular maintenance as Flutter changes. :)

**Comment by nt4f04uNd** [comment]
Hey @Piinks, sorry for taking a long time to respond here. I added a small integration test in recent commit, however, I'm not sure if you wanted it to be integration. Do we want to see that this builds and actually runs, or just that flutter code works (and therefore a regular automated test) would be fine instead?

I submitted a test config here https://github.com/flutter/tests/pull/131, but I'm not sure it will be able to run integration tests.

**Review by Piinks — COMMENTED** [review]
Hey @nt4f04uNd no worries about the delay, happy to see you back.

> Do we want to see that this builds and actually runs, or just that flutter code works (and therefore a regular automated test) would be fine instead?

I think an integration test would work, but both sounds like it could not hurt. Can you add a simple widget test that ensures the Flutter code works?

I can help with the flutter/tests PR once this lands. :)

**Pull request merged**